### PR TITLE
feat: compare time series models

### PR DESCRIPTION
**Pull request by TremaMiguel**
#### Describe the changes you've made

Major changes include:

* `compare_models()` support for time series models.
* `create_model_without_cv` switching X, y order before fitting estimator. [link](https://github.com/pycaret/pycaret/blob/ts_compare_mmodels/pycaret/internal/PycaretExperiment.py#L16449)

Minor changes include:

* change sktime requirement to version `0.5.3`, sktime version `0.6.0` changes functionality in `ExpandingWindowSplitter`.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

No tests added

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: pycaret/internal/PycaretExperiment.py [hunk 1]
```diff
@@ -16206,7 +16215,7 @@ def compare_models(
         fold: Optional[Union[int, Any]] = None,
         round: int = 4,
         cross_validation: bool = True,
-        sort: str = "R2",
+        sort: str = "mape",
```
Comment (ngupta23) [inline pycaret/internal/PycaretExperiment.py:16218]: Could you change this to `smape`?

File: pycaret/internal/PycaretExperiment.py [hunk 2]
```diff
@@ -16261,7 +16272,7 @@ def compare_models(
             is ignored when cross_validation is set to False.
 
 
-        sort: str, default = 'R2'
+        sort: str, default = 'mape'
```
Comment (ngupta23) [inline pycaret/internal/PycaretExperiment.py:16275]: Could you change this to `smape`?

File: pycaret/time_series.py [hunk 3]
```diff
@@ -257,7 +257,7 @@ def compare_models(
     fold: Optional[Union[int, Any]] = None,
     round: int = 4,
     cross_validation: bool = True,
-    sort: str = "R2",
+    sort: str = "mape",
```
Comment (ngupta23) [inline pycaret/time_series.py:260]: Could you change this to `smape'?

**Review by ngupta23 — CHANGES_REQUESTED** [review]
Looks good to me. Just change the default metric to `smape` and resolve the conflict with mt branch and should be good to go.

File: pycaret/internal/PycaretExperiment.py (hunk 1 above)
Comment (TremaMiguel) [reply pycaret/internal/PycaretExperiment.py]: yes

File: pycaret/internal/PycaretExperiment.py [hunk 4]
```diff
@@ -5136,8 +5136,11 @@ def compare_models(
             display.display_monitor()
             display.display_master_display()
 
+        # pull() method retrieves metrics id to column the results
+        id_or_display_name = lambda x: x.id if self._ml_usecase == MLUsecase.TIME_SERIES else x.display_name
```
Comment (Yard1) [inline pycaret/internal/PycaretExperiment.py:5140]: Please turn this into a normal function (with `def`)

**Review by Yard1 — CHANGES_REQUESTED** [review]
Just this one - we should avoid using lambda functions unless they are really tiny

**Review by ngupta23 — APPROVED** [review]

File: pycaret/internal/PycaretExperiment.py (hunk 4 above)
Comment (TremaMiguel) [reply pycaret/internal/PycaretExperiment.py]: refactor to `id_or_display_name` function

**Pull request merged**